core/unix/os.c

```diff
@@ -3719,9 +3719,11 @@ dr_create_client_thread(void (*func)(void *param), void *arg)
 #        endif
     LOG(THREAD, LOG_ALL, 1, "dr_create_client_thread xsp=" PFX " dstack=" PFX "\n", xsp,
         get_clone_record_dstack(crec));
+    os_clone_pre(dcontext);
```

Comment:
Won't the 32-bit CLONE_SETTLS undo this?  I don't think we should be doing both.

Reply:
Again, I think it is "fs". The 32-bit version of the test works fine it seems. Not sure why we are treating 32-bit different using SETTLS? It does seem like os_switch_lib_tls and the SETTLS are redundant in a way, for 32-bit.

Reply:
That is the point: you're adding code for 32-bit which is undone by the existing code.  One or the other should be changed.  Probably the older, to avoid having the correct magic, though I have not thought it through.

Reply:
Yes that's true. I did not understand why SETTLS is used for 32-bit. I felt it is more safe to leave it than changing it. There is nothing I can see where this seems necessary. Clearing the segment register in dynamorio_clone and the SETTLS seem redundant. I felt more comfortable with leaving this, but that said, I removed it, test(s) look(s) fine.